src/NuGet.Services.AzureSearch/SearchService/AzureSearchQuery.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace NuGet.Services.AzureSearch.SearchService
+{
+    /// <summary>
+    /// The type used to generate Azure Search Service queries. Used by <see cref="SearchTextBuilder"/>.
+    /// Given the query "fieldA:value1 value2":
+    /// 
+    ///   * "value1" is a field-scoped value 
+    ///   * "value2" is a non-field scoped value
+    /// </summary>
+    internal class AzureSearchQuery
+    {
+        private static readonly HashSet<char> SpecialCharacters = new HashSet<char>
+        {
+            '+', '-', '&', '|', '!', '(', ')', '{', '}', '[', ']', '^', '"', '~', '*', '?', ':', '\\', '/',
+        };
+
+        private readonly List<string> _nonFieldScopedValues;
+        private readonly Dictionary<string, List<string>> _fieldScopedValues;
+
+        public AzureSearchQuery()
+        {
+            _nonFieldScopedValues = new List<string>();
+            _fieldScopedValues = new Dictionary<string, List<string>>();
+        }
+
+        public void AddNonFieldScopedValues(IEnumerable<string> values)
+        {
+            _nonFieldScopedValues.AddRange(values);
+        }
+
+        public void AddFieldScopedValues(string fieldName, IEnumerable<string> values)
+        {
+            if (!_fieldScopedValues.ContainsKey(fieldName))
+            {
+                _fieldScopedValues[fieldName] = new List<string>();
+            }
+
+            _fieldScopedValues[fieldName].AddRange(values);
+        }
+
+        public override string ToString()
+        {
+            var result = new StringBuilder();
+
+            foreach (var fieldScopedTerm in _fieldScopedValues)
+            {
+                // Separate groups of field-scoped terms are AND'd together.
+                if (result.Length > 0)
+                {
+                    result.Append(" AND ");
```

Comment:
I would recommend choosing between `+` and `AND`.
https://docs.microsoft.com/en-us/azure/search/query-lucene-syntax#and-operator-and--or-

`+` seems a bit more idiomatic.

Reply:
Moved everything to the required operator `+`